src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,188 @@
+1. The failing account has no permission to access a particular resource used in the CloudFormation. 
+1. The application code relies on “secrets” and secrets are not correctly created or copied in the failing environment.
+Here, you would want to debug the failing environment and root cause the problem.
+Assume, you have created a project in a “Demo” environment and are planning to expand it to a full CI/CD environment with Dev, Test and Prod, your project most probably follows a pattern specified in the [Amplify Teams guide](https://docs.amplify.aws/cli/teams/overview/#setting-up-prod-and-dev-environments). However on completion of all steps you observe that Amplify studio displays the “build” step of one of your stages as failed and others as successfully deployed.
+
+##### Suggested Resolution: 
+If your application uses secrets, please refer to https://docs.amplify.aws/cli/function/secrets/#accessing-the-values-in-your-function in the multi-env section to debug any issues related to secrets not being correctly copied across environments. 
+For debugging other problems, please use the Amplify CLI to deploy the application in the failing account using the following steps
+1. amplify pull --appId successfulEnvAppID --envName successfulEnv
+1. amplify env checkout failingEnv
+1. amplify push 
+
+If the push fails, please check the CloudFormation event logs for the failing resource to get detailed information and use this to fix issues in the environment. Visit the CloudFormation service console and search for you application name. e.g. An application named lil deployed in the beta environment would have a CloudFormation stack named amplify-lil-beta-{random chars}. Click on stack name and the Resources tab to view the status and the error reason.
+This figure displays the CloudFormation status logs for a failed resource deployed through Amplify CLI:
+
+###### fig.1.1 Cloudformation Stack Deployment Error
+![ts-guide-cfn-error](/images/ts-guide-cfn-error.png)
+
+### Handling push failures from: Application resources hitting AWS Service Limits
+<Callout information>
+Error Messages: 
+"Limit on the number of resources in a single stack operation exceeded” 
+"Cannot exceed quota for PoliciesPerRole"
+</Callout>
+
+AWS maintains service quotas (formerly called service limits) for each account to help guarantee the availability of AWS resources and prevent accidental provisioning of more resources than needed. If your application contains many resources, especially for particularly big GraphQL schemas with complex connections and resolvers or REST APIs with many resolvers, it’s likely that you might eventually hit AWS Service Limits leading to amplify push failures. 
+To identify if a push failure is because of exceeding [AWS Service level quotas](https://console.aws.amazon.com/servicequotas) please check the AWS console for the CloudFormation deployment/failure logs.
```

Comment:
this link requires being signed in to the console to view. I'd suggest using the link in my above suggestion